src/transformers/utils/quantization_config.py

```diff
@@ -994,6 +995,95 @@ def post_init(self):
             self.linear_weights_not_to_quantize = []
 
 
+@dataclass
+class VptqLayerConfig(QuantizationConfigMixin):
+    """
+    This is used to explain vptq config params for each layer
+    Args:
+        enable_norm: to control if we have scale/bias for fp-weight
+        enable_perm: to perm input_channel or not
+        group_num: how many single groups for vector-quantization
+        group_size:depends on out-features
+        indices_as_float: for Finetuning
+        is_indice_packed: should always be True
+        num_centroids: centriod numbers of clusters
+        num_res_centroids: ditto for residual
+        outlier_size: outliers
+        vector_lens: centroid vector length in quantization
+    """
+
+    def __init__(
+        self,
+        enable_norm: bool = True,
+        enable_perm: bool = True,
+        group_num: int = 1,
+        group_size: int = -1,
+        in_features: int = -1,
+        indices_as_float: bool = False,
+        is_indice_packed: bool = True,
+        num_centroids: tuple = [-1, -1],
+        num_res_centroids: tuple = [-1, -1],
+        out_features: int = -1,
+        outlier_size: int = 0,
+        vector_lens: tuple = [-1, -1],
+        **kwargs,
+    ):
+        self.enable_norm = enable_norm
+        self.enable_perm = enable_perm
+        self.group_num = group_num
+        self.group_size = group_size
+        self.in_features = in_features
+        self.indices_as_float = indices_as_float
+        self.is_indice_packed = is_indice_packed
+        self.num_centroids = num_centroids
+        self.num_res_centroids = num_res_centroids
+        self.out_features = out_features
+        self.outlier_size = outlier_size
+        self.vector_lens = vector_lens
+        self.post_init()
+
+    def post_init(self):
+        r"""
+        Safety checker that arguments are correct
+        """
+        if self.is_indice_packed is False:
+            raise ValueError("is_indice_packed should always be True")
+
+
+@dataclass
+class VptqConfig(QuantizationConfigMixin):
+    """
+    This is a wrapper class about `vptq` parameters.
+
+    Args:
+        enable_proxy_error (`bool`, *optional*, defaults to `False`): calculate proxy error for each layer
+        config_for_layers (`Dict`, *optional*, defaults to `{}`): quantization params for each layer
+        kwargs (`Dict[str, Any]`, *optional*):
+            Additional parameters from which to initialize the configuration object.
+    """
+
```

Comment:
Yes, It's exactly true. Config file contains all specific parameters and it a bit huge.
We used this to customize different quantization parameters for different linear layers to have the best performance.
But we will try to figure out a way to allow a simplified configuration.

Reply:
Can we, for each `Linear` layer like `q_proj` or `down_proj`, have a list of parameter values, and each parameter corresponds to a layer ?

Reply:
Hi @MekkCyber  sorry, not quite following your suggestion, could you please explain a little bit more? 
What we did is for each `linear` layer, we give it a paraments dict, such as
```
model.decoder_model.layers.0.self_attn.q_proj: {"v_len":[-1, 8], .....},
model.decoder_model.layers.0.self_attn.k_proj: {"v_len":[-1, 8], .....},
.
.
.
model.decoder_model.layers.n.self_attn.q_proj: {"v_len":[-1, 8], .....},
```

Reply:
Hi @wejoncy, sorry I wasn't clear, I meant smth like this for example for the `up_proj` linears : 
```
model.layers.mlp.up_proj: {
        "bias": [null, null, ..., null],
        "enable_norm": [true, true, false, ..., true],
        ...
        "num_centroids": [
          (-1, 65536), (-1, 65536), ...,
        ],
        ...
      },
```
So that each property contains a list of values, and each value corresponds to a layer, so the length of the lists, is the number of layers, and for the i-th layer, we access the `bias`property for example using : `model.layers.mlp.up_proj["bias"][i]`